ql/src/java/org/apache/hadoop/hive/ql/parse/ImportSemanticAnalyzer.java

```diff
@@ -772,10 +772,7 @@ we use isExternal and not tableType() method since that always gives type as man
         tgtPath = new Path(table.getDataLocation().toString(),
             Warehouse.makePartPath(partSpec.getPartSpec()));
       } else {
-        Database parentDb = x.getHive().getDatabase(tblDesc.getDatabaseName());
-        tgtPath = new Path(
-            wh.getDefaultTablePath( parentDb, tblDesc.getTableName(), tblDesc.isExternal()),
-            Warehouse.makePartPath(partSpec.getPartSpec()));
+        tgtPath = new Path(getTableDataLocation(wh, tblDesc, x), Warehouse.makePartPath(partSpec.getPartSpec()));
```

Comment:
why don't you store data location into `tablePath` and extract from all the brances method for building the part path
````
tgtPath = new Path(tablePath, Warehouse.makePartPath(partSpec.getPartSpec()))
````

Reply:
yes, we can. I will fix it. Thank you!